src/sci/detci.h

```diff
@@ -0,0 +1,166 @@
+#ifndef _detci_h_
+#define _detci_h_
+
+#include "ci_rdm/ci_rdms.h"
+#include "base_classes/mo_space_info.h"
+#include "helpers/helpers.h"
+#include "integrals/integrals.h"
+#include "base_classes/active_space_method.h"
+#include "base_classes/rdms.h"
+#include "sparse_ci/sparse_ci_solver.h"
+#include "sparse_ci/ci_reference.h"
+#include "integrals/active_space_integrals.h"
+#include "sparse_ci/determinant.h"
+#include "sparse_ci/sigma_vector.h"
+
+namespace forte {
+class DETCI : public ActiveSpaceMethod {
+  public:
+    /**
+     * @brief DETCI Constructor
+     * @param state The state info (symmetry, multiplicity, na, nb, etc.)
+     * @param nroot Number of roots of interests
+     * @param scf_info SCF information
+     * @param options Forte options
+     * @param mo_space_info MOSpaceInfo
+     * @param as_ints Active space integrals
+     */
+    DETCI(StateInfo state, size_t nroot, std::shared_ptr<SCFInfo> scf_info,
+          std::shared_ptr<ForteOptions> options, std::shared_ptr<MOSpaceInfo> mo_space_info,
+          std::shared_ptr<ActiveSpaceIntegrals> as_ints);
+
+    /// Compute the energy
+    double compute_energy() override;
+
+    /// RDMs override
+    std::vector<RDMs> rdms(const std::vector<std::pair<size_t, size_t>>& root_list,
+                           int max_rdm_level) override;
+
+    /// Transition RDMs override
+    std::vector<RDMs> transition_rdms(const std::vector<std::pair<size_t, size_t>>& root_list,
+                                      std::shared_ptr<ActiveSpaceMethod> method2,
+                                      int max_rdm_level) override;
+
+    /// Set options override
+    void set_options(std::shared_ptr<ForteOptions> options) override;
+
+    /// Quiet mode (no printing, for use with CASSCF)
```

Comment:
Should these functions be added to `ActiveSpaceMethods`?

Reply:
Yes, moved to ActiveSpaceMethod